crates/cast/bin/cmd/storage.rs

```diff
@@ -202,23 +202,52 @@ async fn fetch_storage_slots(
     address: NameOrAddress,
     layout: &StorageLayout,
 ) -> Result<Vec<B256>> {
-    // TODO: Batch request
-    let futures: Vec<_> = layout
-        .storage
-        .iter()
-        .map(|slot| {
-            let slot = B256::from(U256::from_str(&slot.slot)?);
-            Ok(provider.get_storage_at(address.clone(), slot.to_ethers(), None))
-        })
-        .collect::<Result<_>>()?;
-
-    join_all(futures).await.into_iter().map(|r| Ok(r?.to_alloy())).collect()
+    let mut results = Vec::new();
+
+    for storage_entry in &layout.storage {
+        let slot_number = U256::from_str_radix(&storage_entry.slot, 10)?.to_be_bytes();
+
+        let storage_result = provider
+            .get_storage_at(address.clone(), B256::from(slot_number).to_ethers(), None)
+            .await?;
+        let storage_bytes = storage_result.as_fixed_bytes();
+
+        let type_details = layout.types.get(&storage_entry.storage_type).ok_or_else(|| {
+            eyre::eyre!("Type details not found for {}", &storage_entry.storage_type)
+        })?;
+
+        let number_of_bytes = type_details.number_of_bytes.parse::<usize>().map_err(|_| {
+            eyre::eyre!("Invalid number of bytes: {}", type_details.number_of_bytes)
+        })?;
```

Comment:
please keep the iterator, because now this requests all slots in sequence and no longer concurretnly